okhttp/src/main/java/io/grpc/okhttp/OkHttpClientTransport.java

```diff
@@ -486,13 +487,23 @@ void onException(Throwable failureCause) {
    */
   private void onError(ErrorCode errorCode, String moreDetail) {
     frameWriter.goAway(0, errorCode, new byte[0]);
-    onGoAway(0, toGrpcStatus(errorCode).augmentDescription(moreDetail));
+    startGoAway(0, toGrpcStatus(errorCode).augmentDescription(moreDetail));
   }
 
-  private void onGoAway(int lastKnownStreamId, Status status) {
-    boolean notifyShutdown;
+  private void startGoAway(int lastKnownStreamId, Status status) {
+    synchronized (lock) {
+      if (goingAway) {
+        // Another go-away is inprocess, ignore this one.
```

Comment:
Fair enough. I don't feel that strong for it. Could you add a comment to `goingAway` to document its semantics?